### PR TITLE
Bump k8s dependecies to v1.25

### PR DESCRIPTION
**Pull request by yaron-idan**
This PR bumps k8s dependencies to v1.25 to address CVE-2020-8558 and close issue #742

### CODE REVIEW

**Review by MartinForReal — COMMENTED** [review]
Please remove files under .idea

**Review by MartinForReal — COMMENTED** [review]
Please remove these files

**Review by MartinForReal — COMMENTED** [review]
Please remove these files

File: pkg/exporters/k8sexporter/problemclient/problem_client.go [hunk 1]
```diff
@@ -107,7 +108,7 @@ func (c *nodeProblemClient) SetConditions(newConditions []v1.NodeCondition) erro
 	if err != nil {
 		return err
 	}
-	return c.client.RESTClient().Patch(types.StrategicMergePatchType).Resource("nodes").Name(c.nodeName).SubResource("status").Body(patch).Do().Error()
+	return c.client.RESTClient().Patch(types.StrategicMergePatchType).Resource("nodes").Name(c.nodeName).SubResource("status").Body(patch).Do(context.TODO()).Error()
```
Comment (MartinForReal) [inline pkg/exporters/k8sexporter/problemclient/problem_client.go:111]: I think we'd better to introduce context package into NPD, context.TODO() should not be used with product code.

File: pkg/systemstatsmonitor/memory_collector_linux.go [hunk 2]
```diff
@@ -38,30 +38,30 @@ func (mc *memoryCollector) collect() {
 	}
 
 	if mc.mBytesUsed != nil {
-		memUsed := meminfo.MemTotal - meminfo.MemFree - meminfo.Buffers - meminfo.Cached - meminfo.Slab
-		mc.mBytesUsed.Record(map[string]string{stateLabel: "free"}, int64(meminfo.MemFree)*1024)
+		memUsed := *meminfo.MemTotal - *meminfo.MemFree - *meminfo.Buffers - *meminfo.Cached - *meminfo.Slab
```
Comment (MartinForReal) [inline pkg/systemstatsmonitor/memory_collector_linux.go:41]: Do you think if we need to make sure pointer is not nil.

File: go.mod [hunk 3]
```diff
@@ -3,89 +3,161 @@ module k8s.io/node-problem-detector
...
-	k8s.io/api v0.0.0-20190816222004-e3a6b8045b0b
-	k8s.io/apimachinery v0.0.0-20190816221834-a9f1d8a9c101
+	github.com/stretchr/testify v1.8.1
+	go.opencensus.io v0.24.0
+	golang.org/x/crypto v0.9.0
+	golang.org/x/oauth2 v0.8.0
+	golang.org/x/sys v0.8.0
+	google.golang.org/api v0.118.0
+	k8s.io/api v0.27.2
+	k8s.io/apimachinery v0.27.2
```
Comment (MartinForReal) [inline go.mod:32]: Please check the staus of go mod graph. Although k8s.io related dependencies

**Review by MartinForReal — CHANGES_REQUESTED** [review]

File: pkg/systemstatsmonitor/memory_collector_linux.go (hunk 2 above)
Comment (yaron-idan) [reply pkg/systemstatsmonitor/memory_collector_linux.go:41]: It's a good idea, I've added a check to the condition.

Comment (MartinForReal) [reply pkg/systemstatsmonitor/memory_collector_linux.go:41]: I mean maybe we should check if meminfo.MemFree, meminfo.Buffers or other item is nil

Comment (yaron-idan) [reply pkg/systemstatsmonitor/memory_collector_linux.go:41]: Gotcha, I updated the condition. Does this fit what you meant?

File: pkg/exporters/k8sexporter/problemclient/problem_client.go (hunk 1 above)
Comment (mmiranda96) [reply pkg/exporters/k8sexporter/problemclient/problem_client.go]: I agree, but let's make that a separate change. I think it's fine to use TODO for now.

File: go.mod (hunk 3 above)
Comment (yaron-idan) [reply go.mod:32]: I haven't found any mention of v1.27 when running `go mod graph`, are you sure it's been required somewhere in the project?

File: pkg/exporters/k8sexporter/problemclient/problem_client.go (hunk 1 above)
Comment (yaron-idan) [reply pkg/exporters/k8sexporter/problemclient/problem_client.go]: I've worked on a commit adding context last night, [here it is](https://github.com/kubernetes/node-problem-detector/pull/761/commits/e31f5a39613d34a8c9825803c7f14a0d9a4688b4) for reference.

**Comment by MartinForReal** [comment]
https://github.com/kubernetes/node-problem-detector/pull/793 is merged.

**Comment by hakman** [comment]
NPD dependencies were bumped to 1.28.2. Most CVEs should be addressed in the next release.

**Pull request closed without merging**